test/integration/createProvider_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"testing"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	crossplaneapis "github.com/crossplaneio/crossplane/apis"
+	"github.com/hasheddan/athodyd"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/crossplaneio/stack-gcp/apis"
+	"github.com/crossplaneio/stack-gcp/apis/v1alpha3"
+	"github.com/crossplaneio/stack-gcp/pkg/controller"
+)
+
+func TestCreateAllClasses(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		test   func(c client.Client) error
+	}{
+		"CreateProvider": {
+			reason: "A GCP Provider should be created without error.",
+			test: func(c client.Client) error {
+				p := &v1alpha3.Provider{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "gcp-provider",
+					},
+					Spec: v1alpha3.ProviderSpec{
+						Secret: runtimev1alpha1.SecretKeySelector{
+							Key: "credentials.json",
+							SecretReference: runtimev1alpha1.SecretReference{
+								Name:      "example-provider-gcp",
+								Namespace: "crossplane-system",
+							},
+						},
+						ProjectID: "crossplane-playground",
+					},
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), p); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), p)
+			},
+		},
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	a, err := athodyd.New(cfg, athodyd.WithCRDDirectoryPaths([]string{"../../config/crd", "../testdata/crossplane"}))
```

Comment:
Perhaps WithCRDDirectoryPaths should take a variadic amount of string arguments rather than a slice? `athodyd. WithCRDDirectoryPaths("../../config/crd", "../testdata/crossplane")` feels a little cleaner and more succinct. I believe you could still use `somestringslice...` to unroll `somestringslice` into string parameters if you needed to.